src/core/libmaven/PeakGroup.cpp

```diff
@@ -1069,3 +1059,91 @@ void PeakGroup::setGroupId(int groupId)
         peak.groupNum = groupId;
     }
 }
+void PeakGroup::setUserLabel(char label)
+{
+    if (label == 'g') {
+        _predictedLabel = ClassifiedLabel::Signal;
+    } else if (label == 'b') {
+        _predictedLabel = ClassifiedLabel::Noise;
+    } else if (label == '\0') {
+        _predictedLabel = ClassifiedLabel::None;
+    } else {
+        // ignore invalid labels
+        return;
+    }
+    _predictionProbability = 0.0f;
+    _predictionInference.clear();
+    _userLabel = label;
+
+    if (parent != nullptr && parent->tagString == C12_PARENT_LABEL) {
+        parent->setUserLabel(label);
+        return;
+    }
+
+    for (auto& child : _childIsotopes) {
+            child->setUserLabel(label);
+    }
```

Comment:
As it is now, here is what I think will happen:
* If you mark an child isotope/adduct as good/bad or unmark it, then its parent will also be marked as good/bad/unmarked, right?
* When the parent's label is being set, the second loop will run and set all its child items to have the same label as the parent.

Are those things what should be happening? I think not. In isotopic labelled studies, you can very easily have a non-parent child group that has higher quality while the parent is a non-existent/bad peak-group.

In any case, you can discuss this with Richa on what should be the expected behavior. In my opinion, only mark the peak-group that is currently selected and leave its parent/siblings as they are.

Reply:
Exactly, I feel the same. Setting the user label is outside peakML. Removing this piece of code too. So, that it gives the same functionality as it was in v0.12